File: src/main/java/it/unibo/crabinv/Model/i18n/SUPPORTED_LOCALES.java
package it.unibo.crabinv.Model.i18n;

import java.util.Locale;

/**
 * Lists all currently supported locales to ensure correct usage
 */
public enum SUPPORTED_LOCALES {
    ENGLISH(new Locale.Builder().setLanguage("en").setRegion("US").build()),
    ITALIAN(new Locale.Builder().setLanguage("it").setRegion("IT").build());

    private final Locale locale;

    SUPPORTED_LOCALES(Locale locale) {
        this.locale = locale;
    }

    public Locale getLocale() {
        return locale;
    }
}

File: src/main/java/it/unibo/crabinv/model/core/i18n/SupportedLocales.kt
package it.unibo.crabinv.model.core.i18n

import java.util.Locale

/**
 * Provides all currently supported locales to ensure correct usage.
 * It also lists the path of the flag representing said language, and it's localised name
 */
enum class SupportedLocales(
    /**
     * @return the locale
     */
    val locale: Locale,
    /**
     * @return the image path of the language flag
     */
    val imagePath: String,
    /**
     * @return the localized name of the language
     */
    val localizedName: String,
) {
    ENGLISH(
        Locale
            .Builder()
            .setLanguage("en")
            .setRegion("US")
            .build(),
        "/flags/flag_en.png",
        "ENGLISH",
    ),
    ITALIAN(
        Locale
            .Builder()
            .setLanguage("it")
            .setRegion("IT")
            .build(),
        "/flags/flag_it.png",
        "ITALIANO",
    ),
    ;

    override fun toString(): String = this.localizedName
}
